services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -214,6 +232,68 @@ class HakemusService(
         return hakemusRepository.save(hakemus).toHakemus()
     }
 
+    /**
+     * Create a new cable report application for an excavation announcement that has a cable report
+     * applied for.
+     *
+     * @return the new cable report application entity or null if no new application was applied for
+     */
+    private fun newJohtoselvitysAppliedFor(hakemus: HakemusEntity): HakemusEntity? {
+        val hakemusEntityData = hakemus.hakemusEntityData
+        val appliedFor =
+            when (hakemusEntityData) {
+                is KaivuilmoitusEntityData -> !hakemusEntityData.cableReportDone
+                else -> return null
+            }
+        if (!appliedFor) {
+            return null
+        }
+        logger.info("Creating a new johtoselvityshakemus for a kaivuilmoitus. id=${hakemus.id}")
```

Comment:
I would encourage logging `${hakemus.logString()}` instead of just the id. Even though the `alluId` and `applicationIdentifier` are always nulls here.

Reply:
You're right. There could be some more information in the logString in the future like status or such. And it makes finding information in logs easier also.